component/src/main/java/org/cbioportal/genome_nexus/component/annotation/IndexResolver.java

```diff
@@ -0,0 +1,158 @@
+package org.cbioportal.genome_nexus.component.annotation;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.mongodb.lang.Nullable;
+
+import org.cbioportal.genome_nexus.model.Index;
+import org.cbioportal.genome_nexus.model.TranscriptConsequence;
+import org.cbioportal.genome_nexus.model.VariantAnnotation;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class IndexResolver {
+    private final ProteinChangeResolver proteinChangeResolver;
+    @Autowired
+    public IndexResolver(ProteinChangeResolver proteinChangeResolver)
+    {
+        this.proteinChangeResolver = proteinChangeResolver;
+    }
+    @NotNull
+    public Index resolve(VariantAnnotation variantAnnotation)
+    {
+        Index index = new Index();
+        index.setVariant(this.resolveVariant(variantAnnotation));
+        index.setHugoSymbol(this.resolveHugoSymbol(variantAnnotation));
+        index.setHgvsp(this.resolveHgvsp(variantAnnotation));
+        index.setHgvsc(this.resolveHgvsc(variantAnnotation));
+        index.setCdna(this.resolveCdna(variantAnnotation));
+        index.setHgvspShort(this.resolveHgvspShort(variantAnnotation));
+
+        return index;
+    }
+
+    @Nullable
+    private String resolveVariant(VariantAnnotation variantAnnotation)
+    {
+        String variant = null;
+
+        if (variantAnnotation != null &&
+            variantAnnotation.getHgvsg() != null)
+        {
+            variant = variantAnnotation.getHgvsg();
+        }
+
+        return variant;
+    }
+
+    @Nullable
+    private List<String> resolveHugoSymbol(VariantAnnotation variantAnnotation)
```

Comment:
Should we move this method into `HugoGeneSymbolResolver` and name it `resolveAllHugoGeneSymbols`?